config/cloudinit/datasource/metadata/metadata.go

```diff
@@ -26,25 +26,26 @@ import (
 )
 
 type Service struct {
-	Root         string
-	Client       pkg.Getter
-	APIVersion   string
-	UserdataPath string
-	MetadataPath string
-	lastError    error
+	Root                 string
+	Client               pkg.Getter
+	IsAvailableCheckPath string
```

Comment:
I think that keeping the `APIVersion` is fine.
Do you have any important reason to change it?

Reply:
The `IsAvailable` implementation used `ms.Root + ms.APIVersion` and made executed a get call to it. This seems to work fine for other providers, but for Cloudstack `ip/latest` results in a HTTP error. That's why for Cloudstack the `IsAvailable` calls `ip/latest/meta-data`, and instead of introducing another parameter I changed this one. But we could add `APIVerison` again and have both parameters.

Reply:
Well I thought about it again and have now introduced a `NewDatasourceWithCheckPath` with an additional parameter. I think that makes it more clear. I've pushed an update, what do you think?